src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/IExtendedQueryTagStore.cs

```diff
@@ -19,9 +19,10 @@ public interface IExtendedQueryTagStore
         /// Notes: once moved to job framework, the return will jobid, to save development effort, just return task for now.
         /// </summary>
         /// <param name="extendedQueryTagEntries">The extended query tag entries.</param>
+        /// <param name="maxAllowedCount">The max allowed count.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The tag key.</returns>
-        Task AddExtendedQueryTagsAsync(IEnumerable<AddExtendedQueryTagEntry> extendedQueryTagEntries, CancellationToken cancellationToken = default);
+        Task AddExtendedQueryTagsAsync(IEnumerable<AddExtendedQueryTagEntry> extendedQueryTagEntries, int maxAllowedCount, CancellationToken cancellationToken = default);
```

Comment:
For IExtendedQueryTagStore perspective, since it's simply a wrapper on storage procedure, I think it make sense to be part of interface.

For IAddExtendedQueryTagService perspective, it's an implementation detail, and value is from config file.

Reply:
Is there a reason the `maxValue` is not simply read by the `IExtendedQueryTagStore` implementation from the config directly as oppose to the service passing it? Is there a need for the service to have this flexibility?